editor/src/viewport_tools/tools/select.rs

```diff
@@ -296,6 +303,63 @@ impl SelectToolData {
 	}
 }
 
+/// Handles the selected edges whilst dragging the layer bounds
+#[derive(Clone, Debug, Default)]
+struct SelectedEdges {
+	bounds: [DVec2; 2],
+	top: bool,
+	bottom: bool,
+	left: bool,
+	right: bool,
+}
+
+impl SelectedEdges {
+	fn new(top: bool, bottom: bool, left: bool, right: bool, bounds: [DVec2; 2], pivot: &mut DVec2) -> Self {
+		// Calculate the pivot for the operation (the opposite point to the edge dragged)
+		*pivot = {
```

Comment:
Is there any other clean way we could do this that doesn't involve "returning data" to the arguments of the caller? This feels too much like a C hack which Rust's superior feature set should help us avoid. If you think this is the cleanest way, no worries.

Reply:
Also similar to C# out parameters :). This probably should have been refactored when I moved the pivot out of the SelectedEdges struct.